packages/main/src/SplitButton.ts

```diff
@@ -275,62 +281,100 @@ class SplitButton extends UI5Element {
 		};
 	}
 
+	/**
+	 * Function that makes sure the focus is properly managed.
+	 *
+	 * @private
+	 */
+	_manageFocus(button?: Button | SplitButton) {
+		const buttons: Array<Button | SplitButton> = [this.textButton!, this.arrowButton!, this];
+
+		buttons.forEach(btn => {
+			if (btn) { btn.focused = btn === button; }
+		});
+	}
+
 	onBeforeRendering() {
 		this._textButtonIcon = this.textButton && this.activeIcon !== "" && (this._textButtonActive) && !this._shiftOrEscapePressed ? this.activeIcon : this.icon;
 		if (this.disabled) {
 			this._tabIndex = "-1";
 		}
 	}
 
+	_handleMouseClick(e: MouseEvent) {
```

Comment:
I think that we can remove the logic for disabled button since the click event won't be fired if button has disabled attribute. Also the check for HTMLSlotElement since we listen for click from ui5-button